deployer/deployer.py

```diff
@@ -297,23 +296,64 @@ def deploy_version_without_eureka(self, version, health_check, health_check_port
             }
         }
 
-        request = self.request("deployment/start", json.dumps(data))
-
-        time.sleep(30)
+        success = self.start_deployment(data, version)
+        if success:
+            self.resize_asg(asg_name=version)
 
-        self.resize_asg(asg_name=version)
+            good_deploy = self.wait_asg_ready(version=version,
+                                              health_check=health_check,
+                                              health_check_port=health_check_port)
 
-        good_deploy = self.wait_asg_ready(version=version,
-                                          health_check=health_check,
-                                          health_check_port=health_check_port)
+            if good_deploy:
+                if remove_old:
+                    self.remove_old_asg(new_asg_name=version)
+            else:
+                self.disable_asg(asg_name=version)
 
-        if good_deploy:
-            if remove_old:
-                self.remove_old_asg(new_asg_name=version)
         else:
-            self.disable_asg(asg_name=version)
+            raise Exception("Fucking asgard!")
+
+    def start_deployment(self, data, version):
+        wait_seconds_after_stack_start = 10
+        for i in range(1, 10):
+            result = self.request("deployment/start", json.dumps(data))
+            id_task = ast.literal_eval(result.content)["deploymentId"]
+            print(id_task)
+            time.sleep(wait_seconds_after_stack_start)
+            success = self.check_task_creation(version, id_task)
+            if success:
+                break
+            else:
+                auto_scaling_check = self.check_for_auto_scaling_group_creation(version)
+                if auto_scaling_check:
+                    raise Exception("Error: Autoscaling is up, but instances not working propertly")
+                else:
+                    print("Autoscaling error")
+        return success
+
+    def check_task_creation(self, version, id_task):
+        retries = 200
+        wait_seconds = 10
+        normalized_version = version[:-5]
+        url = "http://{0}".format(self.asgard_base_url)
+
+        print("Start deploy version: {}".format(normalized_version))
+        for r in range(retries):
+            status = check_asgard_stack_status(url, normalized_version, id_task)
+            print("## MS {} retry: {}".format(status, r))
+            if status == "completed":
+                return True
+            elif status == "failed":
+                return False
+            time.sleep(wait_seconds)
+        return False
 
-        return request.status_code == 200
+    def check_for_auto_scaling_group_creation(self, asg_name):
+        resp = self.request("autoScaling/show/{}.json".format(asg_name), None)
```

Comment:
@Arcontus you don't need to pass the last parameter because it have a default value:
```
       resp = self.request("autoScaling/show/{}.json".format(asg_name))
```